app/controllers/subnets_controller.rb

```diff
@@ -40,21 +40,23 @@ def destroy
 
   # query our subnet dhcp proxy for an unused IP
   def freeip
-    not_found and return unless (s=params[:subnet_id].to_i) > 0
+    invalid_request and return unless (s=params[:subnet_id].to_i) > 0
     organization = params[:organization_id].blank? ? nil : Organization.find(params[:organization_id])
     location = params[:location_id].blank? ? nil : Location.find(params[:location_id])
     Taxonomy.as_taxonomy organization, location do
-      not_found and return unless (subnet = Subnet.authorized(:view_subnets).find(s))
+      unless (subnet = Subnet.authorized(:view_subnets).find(s))
+        render_403 #returns false, cannot put it in a oneliner
```

Comment:
not_found means that the resource is not there, it's not related to the context of the user.
403 means forbidden - which means that under the current context the action is not possible, but if the context will be different, the action will be available.
I think the proper use of not_found will be if the system just can't find available IP, just as I did [three lines below](https://github.com/ShimShtein/foreman/blob/9427/app/controllers/subnets_controller.rb#L54)